test/Microsoft.DotNet.ProjectJsonMigration.Tests/GivenAProjectMigrator.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.Build.Construction;
+using Microsoft.DotNet.ProjectJsonMigration.Rules;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.Tools.Common;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using NuGet.Frameworks;
+using Xunit;
+
+namespace Microsoft.DotNet.ProjectJsonMigration.Tests
+{
+    public class GivenAProjectMigrator : TestBase
+    {
+        [Fact]
+        public void It_copies_ProjectDirectory_contents_to_OutputDirectory_when_the_directories_are_different()
+        {
+            var testProjectDirectory = TestAssetsManager.CreateTestInstance("TestAppSimple", callingMethod: "z")
+                .WithLockFiles().Path;
+            var outputDirectory = Temp.CreateDirectory().Path;
+
+            var projectDirectoryRelativeFilePaths = EnumerateFilesWithRelativePath(testProjectDirectory);
+
+            var mockProj = ProjectRootElement.Create();
+            var testSettings = new MigrationSettings(testProjectDirectory, outputDirectory, "1.0.0", mockProj);
+
+            var projectMigrator = new ProjectMigrator(new FakeEmptyMigrationRule());
+            projectMigrator.Migrate(testSettings);
+
+            foreach (var projectDirectoryRelativeFilePath in projectDirectoryRelativeFilePaths)
+            {
+                File.Exists(Path.Combine(outputDirectory, projectDirectoryRelativeFilePath)).Should().BeTrue();
+            }
+        }
+
+        [Fact]
+        public void It_throws_when_migrating_a_deprecated_projectJson()
```

Comment:
deprecated project detection is currently a set of functionality that is included in the ProjectModel ProjectReader. 

There is a deprecated TestAsset but I couldn't find any usages of it outside of what I'm doing here, so my feeling is no there is not extensive coverage for this in ProjectModel. 

While I agree that may be necessary to do, I think doing additional work on project model is separate from implementing migration, and outside the scope of this PR.